client/app/components/account_config_item.coffee

```diff
@@ -0,0 +1,179 @@
+{li, span, i} = React.DOM
+classer = React.addons.classSet
+
+AccountActionCreator = require '../actions/account_action_creator'
+LayoutActionCreator  = require '../actions/layout_action_creator'
+RouterMixin = require '../mixins/router_mixin'
+
+
+# Line for the mailbox list.
+module.exports = MailboxItem = React.createClass
+    displayName: 'MailboxItem'
+
+    mixins: [
+        RouterMixin
+        React.addons.LinkedStateMixin # two-way data binding
+    ]
+
+    propTypes:
+        mailbox: React.PropTypes.object
+
+
+    getInitialState: ->
+        return {
+            edited: false
+            favorite: @props.favorite
+        }
+
+
+    # Whether it's edit mode or not, it displays widgets to edit mailbox
+    # properties like the name. Otherwise it displays box information and
+    # button to switch to edit mode and button to delete the current mailbox.
+    render: ->
+        pusher = ""
+        pusher += "    " for j in [1..@props.mailbox.get('depth')] by 1
+        key = @props.mailbox.get 'id'
+
+        if @state.favorite
+            favoriteClass = "fa fa-eye mailbox-visi-yes"
+            favoriteTitle = t "mailbox title favorite"
+        else
+            favoriteClass = "fa fa-eye-slash mailbox-visi-no"
+            favoriteTitle = t "mailbox title not favorite"
```

Comment:
I'd put that in a sub function (like you did in other components), especially since it's far from where it is actually used.